### PR TITLE
Group creation via API

### PR DESCRIPTION
**Pull request by enricostano**
When we hit the private `/` in frontend we make the following API requests first thing:
```
GET /me
GET /memberships
```
We have 3 cases:

1. there are no memberships:
 - redirect to `/onboarding`
2. there is only 1 membership
 - redirect to `/groups/{ memberships.first.id }`
 - fetch `GET /groups/:id` from API
3. there are more than 1 memberships
 - redirect to `/groups`
 - fetch `GET /groups` from API

### CODE REVIEW

File: engines/group/app/controllers/group/api/v1/groups_controller.rb [hunk 1]
```diff
@@ -5,31 +5,42 @@ class GroupsController < ApplicationController
...
 
+        # POST api/v1/groups
+        #
         def create
           group = ::Group::Group.new(group_params)
-          group_creation = GroupCreation.new(group, current_user)
-          if group_creation.create
+          group_creator = GroupCreator.new(group, current_user)
+          if group_creator.create
+            @side_effects << group_creator.side_effects
```
Comment (andresgutgon) [inline engines/group/app/controllers/group/api/v1/groups_controller.rb]: I don't like side effects :) It sounds like a bad thing. This is a resource link. Why we just call `resource_links`?

File: engines/group/app/controllers/group/api/v1/groups_controller.rb [hunk 2]
```diff
@@ -55,10 +68,34 @@ def group_params
...
+          response.headers['Link'] = links
+        end
+
+        def api_link(side_effect)
+          "#{web_app_url}/api/v1/#{model_type(side_effect).downcase.pluralize}/#{side_effect.id}"
+        end
+
+        def model_type(side_effect)
+          side_effect.class.name
+        end
```
Comment (andresgutgon) [inline engines/group/app/controllers/group/api/v1/groups_controller.rb]: Maybe we can wait to add `side_effects || resource_links` in another iteration. I think for onboarding we don't need it. We don't need merbership of the user. 

The next think after create group is bulk invite. When we post to `/invitations/bulk` we check by user_id if user can or not bulk invite. So it's not necessary return membership on group creation. 

Let's talk offline. Maybe I'm missing something

File: engines/group/app/controllers/group/api/v1/groups_controller.rb (hunk 1 above)
Comment (enricostano) [inline engines/group/app/controllers/group/api/v1/groups_controller.rb]: it actually is a bad thing :smiling_imp:

File: engines/group/app/controllers/group/api/v1/groups_controller.rb [hunk 3]
```diff
@@ -5,31 +5,39 @@ class GroupsController < ApplicationController
...
+            @side_effects << group_creator.side_effects
             render json: GroupSerializer.new(group)
           else
             render(
               status: :bad_request,
-              json: {
-                model: group.class.name,
-                errors: group.errors.full_messages
-              }
+              json: group.errors.to_json
```
Comment (andresgutgon) [inline engines/group/app/controllers/group/api/v1/groups_controller.rb]: This is the format I'm using in Signup Complete. Just an object with errors. No need for type

**Pull request merged**